build.gradle.kts

```diff
@@ -115,4 +122,43 @@ apiValidation {
    * Flag to programmatically disable compatibility validator
    */
   validationDisabled = false
-}
\ No newline at end of file
+}
+
+val detektAll by tasks.registering(io.gitlab.arturbosch.detekt.Detekt::class) {
+  configureDetekt(this)
+}
+
+val detektFixAll by tasks.registering(io.gitlab.arturbosch.detekt.Detekt::class) {
+  configureDetekt(this)
+  autoCorrect = true
+}
+
+fun configureDetekt(detekt: io.gitlab.arturbosch.detekt.Detekt) {
+  detekt.apply {
+    description = "Runs over whole code base without the starting overhead for each module."
+    parallel = true
+    buildUponDefaultConfig = true
+    setSource(files(rootDir))
+    config.from(files("$rootDir/detekt.yml"))
+    allRules = false
+    include("**/*.kt")
+    include("**/*.kts")
+    exclude("**/res/**")
+    exclude("**/build/**")
+    val reportsDir = "${rootProject.buildDir}/reports"
+    reports {
+      xml {
+        enabled = true
+        destination = file("$reportsDir/detekt.xml")
+      }
+      html {
+        enabled = true
+        destination = file("$reportsDir/detekt.html")
+      }
+      custom {
+        reportId = "CustomJsonReport"
+        destination = file("$reportsDir/detekt.json")
+      }
+    }
+  }
+}
```

Comment:
if possible avoid configuring plugins in the root build.gadle.kts. This will require projects that consume this as a submodule to additionally configure it their root build.gradle.kts

Reply:
done